Pretty results of dump() for Syntax

Kuniwak opened the pull request:
The result of dump() with Syntax is not useful. For example,
the dumped result of `Tests/SwiftSyntaxTest/Inputs/closure.swift` is the following:

```
▿ // A closure without a signature. The test will ensure it stays the same after
// applying a rewriting pass.
let x: () -> Void = {}
  ▿ data: SwiftSyntax.SyntaxData
    - parent: nil
    ▿ absoluteRaw: SwiftSyntax.AbsoluteRawSyntax
      - raw: // A closure without a signature. The test will ensure it stays the same after
// applying a rewriting pass.
let x: () -> Void = {} #0
        ▿ super: Swift.ManagedBuffer<SwiftSyntax.RawSyntaxBase, Swift.UInt64>
          ▿ header: SwiftSyntax.RawSyntaxBase
	  ...
```

This patch improves this result to the following:

```
▿ SwiftSyntax.SourceFileSyntax
  ▿ statements: SwiftSyntax.CodeBlockItemListSyntax
    ▿ SwiftSyntax.CodeBlockItemSyntax
      ▿ item: SwiftSyntax.VariableDeclSyntax
        - attributes: nil
        - modifiers: nil
        ▿ letOrVarKeyword: SwiftSyntax.TokenSyntax
          - text: "let"
          ▿ leadingTrivia: SwiftSyntax.Trivia
            ▿ pieces: 4 elements
              ▿ TriviaPiece
                - lineComment: "// A closure without a signature. The test will ensure it stays the same after"
                ...
```

Kuniwak commented:
@swift-ci Please test

harlanhaskins commented on Tests/LinuxMain.swift:
```diff
@@ -16,6 +16,7 @@ XCTMain({ () -> [XCTestCaseEntry] in
     testCase(TokenSyntaxTestCase.allTests),
     testCase(SyntaxTreeModifierTests.allTests),
     testCase(TriviaTests.allTests),
+    testCase(CustomRelectableTests.allTests),
```
Typo: `CustomReflectableTests`.

Kuniwak replied:
Fixed by [494385b](/apple/swift-syntax/pull/94/commits/494385b0ca64d2bfcd2fb5f9721cd5e858563624)

Kuniwak commented:
I'm working for necessary tests.

harlanhaskins commented:
@swift-ci please test

Kuniwak commented:
I added the tests for full output of a Syntax node against the expected string at [1ff1a22](/apple/swift-syntax/pull/94/commits/1ff1a22ebaf54321b0bfdac0505e12e69a720f44).

Then, I found type names without the thier module name. But it was not intended so I also fixed it at [5fc0f00](/apple/swift-syntax/pull/94/commits/5fc0f00c6af94bfd3e426ad05f922cfb0bc48c4c).

rintaro commented:
@Kuniwak Thank you for your patience. Could you resolve the conflict?

Kuniwak commented:
I'm reading [the diff](https://github.com/apple/swift-syntax/compare/eb622fa2b19ada77f690fa2235b9ad3b5b14ee3e...ce8fa85e5ec8fd7f6d18337e9bb9f40eb6b107b4) for resolving the conflict.

Kuniwak commented:
@rintaro I resolved the conflict.

rintaro commented:
@swift-ci please test

harlanhaskins commented:
Thanks again, @Kuniwak!
